dashboard-widgets.el

```diff
@@ -213,14 +264,40 @@ predicate value."
       (funcall dashboard-display-icons-p)
     dashboard-display-icons-p))
 
+(defun dashboard-icon-for-dir (dir &rest args)
+  "Get the formatted icon for DIR.
+ARGS should be a plist containing `:height', `:v-adjust',
+or `:face' properties."
+  (cond ((eq dashboard-icon-type 'all-the-icons) (apply #'all-the-icons-icon-for-dir dir args))
+        ((eq dashboard-icon-type 'nerd-icons) (apply #'nerd-icons-icon-for-dir dir args))))
+
+(defun dashboard-icon-for-file (file &rest args)
+  "Get the formatted icon for FILE.
+ARGS should be a plist containing `:height', `:v-adjust',
+or `:face' properties."
+  (cond ((eq dashboard-icon-type 'all-the-icons) (apply #'all-the-icons-icon-for-file file args))
+        ((eq dashboard-icon-type 'nerd-icons) (apply #'nerd-icons-icon-for-file file args))))
+
+(defun dashboard-octicon (name &rest args)
+  "Get the formatted octicon.
+ARGS should be a plist containing `:height', `:v-adjust',
+or `:face' properties."
+  (cond ((eq dashboard-icon-type 'all-the-icons) (apply #'all-the-icons-octicon name args))
+        ((eq dashboard-icon-type 'nerd-icons) (apply #'nerd-icons-octicon name args))))
```

Comment:
And maybe all these places, use `pcase` for consistency. :D

Reply:
oh yeah I should do a search for all of the occurrences. Sorry about that! 

BTW is there any advantage of using `pcase` over `cond`? or it doesn't matter and we just want consistency here?

Reply:
It's a matter of taste. I prefer `pcase` since the code is cleaner. The downside of this, it will add up startup time due to loading the `pcase.el` module.

Reply:
It's a matter of taste. I prefer `pcase` since the code is cleaner. The downside of this, it will add up startup time due to loading the `pcase.el` module.